File: perl/eg/led.pl
use lib "..";
use ev3;

print "Waiting the EV3 brick online...\n";
ev3::ev3_init() > 0 || exit( 1 );

my ( $ok, $state ) = ev3::get_led_brightness( $ev3::EV3_LEFT_RED );
if ( $ok ) {
    print "get_led_brightness( EV3_LEFT_RED ) = $state\n";
} else {
    print "ERROR: get_led_brightness()\n";
}
if ( !ev3::set_led_brightness( $ev3::EV3_LEFT_RED, 0 )) {
    print "ERROR: set_led_brightness()\n";
}
$state = ev3::get_led_trigger_inx( $ev3::EV3_RIGHT_RED );
print "get_led_trigger_inx( EV3_RIGHT_RED ) = $state\n";

if ( !ev3::set_led_trigger( $ev3::EV3_RIGHT_RED, "heartbeat" )) {
    print "ERROR: set_led_trigger()\n";
}
ev3::ev3_uninit();

File: ruby/eg/led.rb
# -*- coding: utf-8 -*-

#FIXME: not the best way to import the library
$LOAD_PATH << if ENV[ 'OS' ] == 'Windows_NT' then '../mingw' else '..' end
require 'ev3'
include Ev3

puts 'Waiting the EV3 brick online...'
if ev3_init() < 1 then exit( 1 ) end

ok, state = get_led_brightness( EV3_LEFT_RED )
if ok
  puts "get_led_brightness( EV3_LEFT_RED ) = #{state}"
else
  puts 'ERROR: get_led_brightness()'
end

if not set_led_brightness( EV3_LEFT_RED, 0 )
  puts 'ERROR: set_led_brightness()'
end

state = get_led_trigger_inx( EV3_RIGHT_RED )
puts "get_led_trigger_inx( EV3_RIGHT_RED ) = #{state}"

if not set_led_trigger( EV3_RIGHT_RED, 'heartbeat' )
  puts 'ERROR: set_led_trigger()'
end

ev3_uninit()
